server/src/main/java/org/elasticsearch/search/aggregations/metrics/PercentileRanksAggregationBuilder.java

```diff
@@ -80,7 +79,7 @@
     static {
         PARSER = new ConstructingObjectParser<>(PercentileRanksAggregationBuilder.NAME, false,
             (a, context) -> new PercentileRanksAggregationBuilder(context, (List) a[0]));
-        ValuesSourceParserHelper.declareNumericFields(PARSER, true, false, false);
+        ValuesSourceParserHelper.declareAnyFields(PARSER, true, true);
```

Comment:
Can we try to be less lenient and require a field that is either numeric or a histogram?

Reply:
Yes, that makes sense and that is the tricky part. @not-napoleon I might want to borrow a bit of your thoughts on how to do this in the best way possible.

Reply:
The parser doesn't currently support that, and right now the convention is to use ANY for aggregations that can accept more than one type of input.  Cleaning that up is a goal of the ValuesSource refactor effort I'm currently working on (see #42949)

Reply:
Thanks @not-napoleon, let's keep this TODO for later then @iverase ?

Reply:
Agreed. I think to be able to support that we require the effort that @not-napoleon is working on (thanks for taking a look!).